server/handlers/policy_relationship_handler.go

```diff
@@ -37,8 +41,45 @@ func (h *Handler) GetRegoPolicyForDesignFile(
 		return
 	}
 
+	name := mux.Vars(r)["name"]
+	typ := mux.Vars(r)["model"]
+	limitstr := r.URL.Query().Get("pagesize")
+	var limit int
+	if limitstr != "all" {
+		limit, _ = strconv.Atoi(limitstr)
+		if limit == 0 { //If limit is unspecified then it defaults to 25
+			limit = DefaultPageSizeForMeshModelComponents
+		}
+	}
+	pagestr := r.URL.Query().Get("page")
+	page, _ := strconv.Atoi(pagestr)
+	if page == 0 {
+		page = 1
+	}
+	offset := (page - 1) * limit
+
+	res, c := h.registryManager.GetEntities(&v1alpha1.PolicyFilter{
+		Kind:       name,
+		SubType:    r.URL.Query().Get("subtype"),
+		Version:    r.URL.Query().Get("version"),
+		ModelName:  typ,
+		APIVersion: r.URL.Query().Get("apiversion"),
+		Limit:      limit,
+		Offset:     offset,
+		OrderOn:    r.URL.Query().Get("order"),
+		Sort:       r.URL.Query().Get("sort"),
+	})
+
+	if *c == 0 {
+		h.log.Error(ErrNoPoliciesFound(fmt.Errorf("0 policies found for given filters")))
+		http.Error(rw, ErrNoPoliciesFound(fmt.Errorf("0 policies found for given filters")).Error(), http.StatusInternalServerError)
+		return
+	}
+	var policy v1alpha1.PolicyDefinition
+	policy, _ = res[0].(v1alpha1.PolicyDefinition)
+
 	// evaluate all the rego policies in the policies directory
-	networkPolicy, err := policies.RegoPolicyHandler(context.Background(), []string{"../meshmodel/policies"}, "data.network_policy", body)
+	networkPolicy, err := policies.RegoPolicyHandler(context.Background(), policy.Expression, "data.network_policy", body)
```

Comment:
since `policy.expression` is loaded dynamically via json, should the query be loaded the same way as well? 

If the client is saying to execute policy on model `kubernetes`, then the query string would be changed based on what is required from that policy, in short, the queries would different for different policies.

When we were using the approach of filesystem, all the rego policies are loaded by the OPA, and we just had to provide the array of queries we need, so, it was possible for client to ask for the query and it will automatically run all the policies from fs and return the response. Now, we might need to load them some different way.